pkg/app/updatedisc/discovery_updater.go

```diff
@@ -52,7 +52,9 @@ func (u *serviceUpdater) Start() {
 
 	u.wg.Add(1)
 	go func() {
-		u.client.UpdateLoop(ctx, u.interval)
+		if err := u.client.Update(ctx); err != nil {
+			return
```

Comment:
May the return here lead to the waitgroup never finishing?

Reply:
Your right, but waitgroup remove complete.